versioned/spi/src/main/java/org/projectnessie/versioned/RefLog.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned;
+
+import java.util.List;
+import org.immutables.value.Value;
+
+/** Represents a reflog-entry stored in the database. */
+@Value.Immutable
+public interface RefLog {
+
+  /** Reflog id of the current entry. */
+  Hash getRefLogId();
+
+  /** Reference on which current operation is executed. */
+  String getRefName();
+
+  /** Reference type can be 'Branch' or 'Tag'. */
+  String getRefType();
+
+  /** Output commit hash of the operation. */
+  Hash getCommitHash();
+
+  /** Parent reflog id of the current entry. */
+  List<Hash> getParents();
```

Comment:
Refactor to `Hash getParent()` (that list is internal to the database-adapter - nothing for the version-store)

Reply:
done, Added one more interface. I was reusing it to avoid extra object.